effectful-core/effectful-core.cabal

```diff
@@ -51,6 +51,7 @@ common language
                         TypeApplications
                         TypeFamilies
                         TypeOperators
+                        AllowAmbiguousTypes
```

Comment:
This is not a good extension to enable globally.

Reply:
Sure, I can remove it all together and add `Proxy` to the place where it is needed.

Reply:
It might be worth experimenting with injective type families, something like

```haskell
data Dispatch = Dynamic | Static

type family DispatchOf (e :: Effect) :: Dispatch

type family AdapterFor (d :: Dispatch) = (r :: Effect -> Type) | r -> d where
  AdapterFor 'Dynamic = HandlerA
  AdapterFor 'Static = DataA

type Adapter e = AdapterFor (DispatchOf e) e

getEnv_
  :: forall e es. e :> es
  => Env es
  -> IO (Adapter e)
getEnv = ...

etc.
```

Then you might not need ambiguous types at all.

Reply:
They aren't actually needed, even in this PR as is, must be a remnant from the past.